python/triqs_dft_tools/sumk_dft_transport.py

```diff
@@ -0,0 +1,854 @@
+
+        sum_k = read_transport_input_from_hdf(sum_k)
+
+    elif code in ('wannier90'):
+        required_entries = ['seedname', 'nk_optics']
+        assert all(entry in w90_params for entry in required_entries), 'Please provide additional keywords "seedname" and "nk_optics" for "code = "wannier90""'
+        # check if spin-unpolarized
+        assert n_inequiv_spin_blocks == 1, "Spin-polarized optical conductivity calculations not implemented with Wannier90"
+
+        # check some of the input
+        pathname = w90_params['pathname'] if 'pathname' in w90_params else './'
+        assert all(isinstance(name, str) for name in ['seedname', 'pathname']), f'Check pathname {w90_params["pathname"]} and seedname {w90_params["seedname"]}'
+        for file_ending in ['.wout', '_hr.dat', '.chk', '.mmn', '.eig']:
+            filename = [pathname, w90_params['seedname'], file_ending]
+            assert os.path.isfile(''.join(filename)), f'Filename {"".join(filename)} does not exist!'
+        calc_velocity = w90_params['calc_velocity'] if 'calc_velocity' in w90_params else True
+        calc_inverse_mass = w90_params['calc_inverse_mass'] if 'calc_inverse_mass' in w90_params else False
+        assert all(isinstance(name, bool) for name in [calc_velocity, calc_inverse_mass]), f'Parameter {calc_velocity} or {calc_inverse_mass} not bool!'
+
+        # select contributions to be used
+        oc_select = w90_params['oc_select'] if 'oc_select' in w90_params else 'both'
+        assert oc_select in ['intra', 'inter', 'both'], '"oc_select" needs to be either ["intra", "inter", "both"]'
+        # gauge choice options 'h' for Hamiltonian and 'w' for Wannier
+        oc_basis = w90_params['oc_basis'] if 'oc_basis' in w90_params else 'h'
+        assert oc_basis in ['h', 'w'], '"oc_basis" needs to be either ["h", "w"]'
+        # finally, make sure oc_select is 'both' for oc_basis = 'w'
+        if oc_basis == 'w' and oc_select != 'both':
+            warn(f'"oc_select" must be "both" for "oc_basis" = "w"!')
+            oc_select = 'both'
+        # further checks for calc_inverse_mass
+        if calc_inverse_mass:
+            assert oc_basis == 'h', '"calc_inverse_mass" only implemented for "oc_basis" == "h"'
+            assert oc_select == 'both', '"oc_select" not implemented for "calc_inverse_mass"'
+        # print some information
+        mpi.report(f'{"Basis choice [h (Hamiltonian), w (Wannier)]:":<60s} {oc_basis}')
+        mpi.report(f'{"Contributions from [intra(-band), inter(-band), both]:":<60s} {oc_select}')
+
+        # recompute sum_k instances on denser grid
+        sum_k, _ = recompute_w90_input_on_different_mesh(sum_k, w90_params['seedname'], nk_optics=w90_params['nk_optics'], pathname=pathname,
+                                                                      calc_velocity=calc_velocity, calc_inverse_mass=calc_inverse_mass, oc_select=oc_select, oc_basis=oc_basis)
+
+    # k-dependent-projections.
```

Comment:
what about disentanglement? Should we somehow build in a check for that, or does this also work with wannierberri?